generator/transit_generator.cpp

```diff
@@ -321,7 +245,182 @@ void BuildTransit(string const & mwmDir, string const & countryId,
+  CHECK(infoGetter, ());
+
+  auto const countryFileGetter = [&infoGetter](m2::PointD const & pt) {
+    return infoGetter->GetRegionCountryId(pt);
+  };
+
+  auto const getMwmRectByName = [&](string const & c) -> m2::RectD {
+    CHECK_EQUAL(countryId, c, ());
+    return infoGetter->GetLimitRectForLeaf(c);
+  };
+
+  CHECK_EQUAL(index.GetMwmId().GetInfo()->GetType(), MwmInfo::COUNTRY, ());
+  auto numMwmIds = make_shared<NumMwmIds>();
+  numMwmIds->RegisterFile(CountryFile(countryId));
+
+  // Note. |indexRouter| is valid while |index| is valid.
+  IndexRouter indexRouter(VehicleType::Pedestrian, false /* load altitudes */,
+                          CountryParentNameGetterFn(), countryFileGetter, getMwmRectByName,
+                          numMwmIds, MakeNumMwmTree(*numMwmIds, *infoGetter),
+                          traffic::TrafficCache(), index.GetIndex());
+
+  // Looking for the best segment for every gate.
+  for (auto & gate : m_gates)
+  {
+    // Note. For pedestrian routing all the segments are considered as twoway segments so
+    // IndexRouter.FindBestSegment() method finds the same segment for |isOutgoing| == true
+    // and |isOutgoing| == false.
+    Segment bestSegment;
+    try
+    {
+      // @todo(bykoianko) For every call of the method below WorldGraph is created. It's not
+      // efficient to create WorldGraph for every gate. The code should be redesigned.
+      if (indexRouter.FindBestSegmentInSingleMwm(gate.GetPoint(),
+                                                 m2::PointD::Zero() /* direction */,
+                                                 true /* isOutgoing */, bestSegment))
+      {
+        CHECK_EQUAL(bestSegment.GetMwmId(), 0, ());
+        gate.SetBestPedestrianSegment(SingleMwmSegment(
+            bestSegment.GetFeatureId(), bestSegment.GetSegmentIdx(), bestSegment.IsForward()));
+      }
+    }
+    catch (MwmIsNotAliveException const & e)
+    {
+      LOG(LERROR, ("Point of a gate belongs doesn't belong to any mwm according to "
+                   "packed_polygons.bin. Gate:", gate, e.what()));
+    }
+  }
+}
+
+bool GraphData::IsUnique() const
+{
+  IsUniqueVisitor v;
+  Visit(v);
+  return v.IsUnique();
+}
+
+bool GraphData::IsSorted() const
+{
+  IsSortedVisitor v;
+  Visit(v);
+  return v.IsSorted();
+}
+
+void DeserializeFromJson(OsmIdToFeatureIdsMap const & mapping,
+                         string const & transitJsonPath, GraphData & data)
+{
+  Platform::EFileType fileType;
+  Platform::EError const errCode = Platform::GetFileType(transitJsonPath, fileType);
+  CHECK_EQUAL(errCode, Platform::EError::ERR_OK, ("Transit graph not found:", transitJsonPath));
+  CHECK_EQUAL(fileType, Platform::EFileType::FILE_TYPE_REGULAR,
+              ("Transit graph not found:", transitJsonPath));
+
+  string jsonBuffer;
+  try
+  {
+    GetPlatform().GetReader(transitJsonPath)->ReadAsString(jsonBuffer);
+  }
+  catch (RootException const & ex)
```

Comment:
Why RootException here?

Reply:
It's for case if a file is absent. As far as I'm aware GetReader() may raise `FileAbsentException`. But I'm not sure about others. 

It's a generator code. Let's catch only `FileAbsentException`. If we miss some exceptions the generator will terminate and we know about that.